File: arduinobot_ws/src/arduinobot_test/scripts/simple_timer.py
#!/usr/bin/env python3
import rospy


def timerCallback(event=None):
    rospy.loginfo('Called timerCallback Function')


if __name__ == '__main__':
    rospy.init_node('timer_node', anonymous=True)

    timer_duration = rospy.Duration(1)
    rospy.Timer(timer_duration, timerCallback)

    rospy.spin()

File: arduinobot_ws/src/arduinobot_test/src/simple_timer.cpp
#include "ros/ros.h"


void timerCallback(const ros::TimerEvent& event)
{
    ROS_INFO("Called timerCallback Function");

}


int main(int argc, char **argv)
{
    ros::init(argc, argv, "timer_node");
    ros::NodeHandle nh;

    ros::Duration timer_duration(1);
    ros::Timer timer = nh.createTimer(timer_duration, timerCallback);

    ros::spin();

    return 0;
}